semgrep_output_v1.atd

```diff
@@ -1680,6 +1680,46 @@ type ci_scan_complete_stats = {
     <python repr="dict">
     <ts repr="map">
     option;
+
+  (* since 1.98.0 *)
+  (* In collaboration with the Data Science team, it was suggested
+   * that we start to group stats by product for organizational purposes.
+   *)
+  ?supply_chain_stats: supply_chain_stats option;
+}
+
+type resolution_method <ocaml attr="deriving show"> <python decorator="dataclass(frozen=True, order=True)"> = [
+  | LockfileParsing 
+  | DynamicResolution 
+]
+
+type dependency_resolution_stats = {
+  resolution_method: resolution_method;
+  dependency_count: int;
+  ecosystem: ecosystem;
+}
+
+type dependency_source_file_kind <ocaml attr="deriving show"> <python decorator="dataclass(frozen=True)"> = [
+  | Lockfile of lockfile_kind
+  | Manifest of manifest_kind
+] 
+
+type dependency_source_file = {
+  kind: dependency_source_file_kind;
+  path: fpath;
+}
+
+type subproject_stats = {
+  (* Unique identifier for the subproject *)
+  subproject_id: string;
```

Comment:
Might be worth describing the scope within which this ID will be unique - if we do the hash-of-all-dependency-source-files then it should be consistent across scans and unique per-repo, right?

Reply:
> then it should be consistent across scans and unique per-repo, right?

Hmmmm, for the most part it should be consistent across scans. The only problem is when a dependency source file is added, renamed, or removed. But I think that's okay since changes are infrequent?

Reply:
Guess you could also make the argument different set of dependency source files == different subproject.

Reply:
updated in [c2cf640](https://github.com/semgrep/semgrep-interfaces/pull/313/commits/c2cf6408bf4af4ac7221a9c6e46d4da41622bdaa)